### PR TITLE
[closes #90] new crop: Asparagus

### LINKED ISSUE #90: new crop Asparagus
add a new crop, asparagus! asparagus is a slow growing crop, but it sells for a high value. seeds are also not that cheap.

### PR DESCRIPTION
**Pull request by lstebner**
new crop: Asparagus. takes 9 days to grow, and unlocks at level 24.

## growth stages

| name | days |
|-------|-----|
| seed | 4 |
| growing | 5 |
| total | 9 |

## some images:

slightly incorrect images as the SPROUT stage was removed after these screenshots and pulled out into a separate feature request (#107)

Screen Shot 2021-06-05 at 11 06 22 AM
Screen Shot 2021-06-05 at 11 18 28 AM
Screen Shot 2021-06-05 at 11 09 31 AM
Screen Shot 2021-06-05 at 11 09 49 AM

### CODE REVIEW

**Comment by jeremyckahn** [comment]
I love it! I will make some for a proper review in the coming days once I have some time. Great work so far! 🙂

File: src/enums.js [hunk 1]
```diff
@@ -80,11 +81,7 @@ export const itemType = enumify([
  * @property farmhand.module:enums.fertilizerType
  * @enum {string}
  */
-export const fertilizerType = enumify([
-  'NONE',
-  'STANDARD',
-  'RAINBOW',
-])
+export const fertilizerType = enumify(['NONE', 'STANDARD', 'RAINBOW'])
```
Comment (lstebner) [inline src/enums.js:84]: i think eslint did this.. is that expected though?

File: src/data/items.js [hunk 2]
```diff
@@ -99,6 +100,32 @@ export const carrot = crop({
...
+  cropType: ASPARAGUS,
+  cropTimetable: {
+    [SEED]: 4,
+    [SPROUT]: 1,
+    [GROWING]: 5,
+  },
+  growsInto: 'asparagus',
+  id: 'asparagus-seed',
+  name: 'Asparagus Seed',
+  tier: 1,
```
Comment (jeremyckahn) [inline src/data/items.js:117]: Let's make this tier 4 (and move it below the other crops in this file). Similarly to what I mentioned in [this comment](https://github.com/jeremyckahn/farmhand/issues/104#issuecomment-855300761), I'd like each tier to only have three crops.

**Review by jeremyckahn — CHANGES_REQUESTED** [review]
This PR is off to a great start! I really like the art design for the Asparagus. I did some manual regression testing and it seems that this is breaking every non-Asparagus crop's growth cycle because of the new `SPROUT` stage. The test failures are legitimate and need to be fixed before this can be merged.

To give a little more detail, Farmhand's implementation is currently designed around the assumption that there are only `SEED`, `GROWING`, and `GROWN` crop stages, so we'll need to account for some crops having a `SPROUT` stage and some not. The easiest solution would be descope this PR and only have the `SEED`, `GROWING`, and `GROWN` Asparagus states, but I'll leave it up to you @lstebner to decide how you'd like to proceed with it.

**Comment by lstebner** [comment]
thanks for the detailed explanation @jeremyckahn . sounds like that's a later issue indeed. i see no reason to hold up asparagus for a new growth stage. I added that mostly out of curiosity. we can discuss if we would like to add a new growth stage separately. i will clean this PR up

File: src/data/items.js (hunk 2 above)
Comment (lstebner) [reply src/data/items.js]: updated

**Comment by lstebner** [comment]
all set for review @jeremyckahn . i left the images because i don't think it hurts anything, but can remove if you prefer.

File: src/data/levels.js [hunk 3]
```diff
@@ -54,6 +54,10 @@ levels[22] = {
   unlocksShopItem: items.tomatoSeed.id,
 }
 
+levels[24] = {
+  unlocksShopItem: items.asparagusSeed.id,
```
Comment (jeremyckahn) [inline src/data/levels.js:58]: 💯

File: src/img/index.js [hunk 4]
```diff
@@ -12,6 +12,10 @@ import jackolantern from './items/jackolantern.png'
 import rainbowCheese from './dishes/rainbow-cheese.png'
 import spaghetti from './dishes/spaghetti.png'
 
+import asparagus from './items/asparagus.png'
+import asparagusGrowing from './items/asparagus-growing.png'
+import asparagusSeed from './items/asparagus-seed.png'
+import asparagusSprout from './items/asparagus-sprout.png'
```
Comment (jeremyckahn) [inline src/img/index.js:18]: Let's remove this to keep it out of the app bundle.

File: src/img/index.js [hunk 5]
```diff
@@ -88,6 +92,10 @@ export const craftedItems = {
 
 export const items = {
   // Crops
+  asparagus,
+  'asparagus-growing': asparagusGrowing,
+  'asparagus-seed': asparagusSeed,
+  'asparagus-sprout': asparagusSprout,
```
Comment (jeremyckahn) [inline src/img/index.js:98]: This line should be removed as well.

File: src/img/items/asparagus-sprout.piskel [hunk 6]
```diff
@@ -0,0 +1 @@
+{"modelVersion":2,"piskel":{"name":"asparagus-sprout","description":"","fps":12,"height":24,"width":24,"layers":["{\"name\":\"Layer 1\",\"opacity\":1,\"frameCount\":1,\"chunks\":[{\"layout\":[[0]],\"base64PNG\":\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAAqElEQVRIS2NkoDFgpLH5DKMWEAxhnEGkbiXx/+axFxQHIVYDQIYbecsznNv6kIGQJSC1IG/gUodhAUiDu6YXw2ulqwQtgKkFWbDz+jaslmC1AOR6ECDkA2IcQ7EFhByDMw7whSss6cDiCp9vKU4lJEcywYRNogKCPsDnQkKuB7kFrwX4UgkxSZQoC3ClEmIieARYAPIiTSOZxBSJVTnBZEqpJaMWEAxBABktlhmV7c8CAAAAAElFTkSuQmCC\"}]}"]}}
```
Comment (jeremyckahn) [inline src/img/items/asparagus-sprout.piskel:1]: Let's definitely keep this in for future use! 🙂

**Review by jeremyckahn — CHANGES_REQUESTED** [review]
This is looking and working great! There's just a few small cleanup items I'd like to get addressed and this will be ready to merge. Thanks for sticking with this!

File: src/img/index.js (hunk 4 above)
Comment (lstebner) [reply src/img/index.js]: good eye! removed the sprout refs from this file

**Review by jeremyckahn — APPROVED** [review]
LGTM! Thanks for adding a new crop and iterating on this PR. Feel free to merge when you're ready and run a `patch` release to ship it to players.

Nice work! 🎉

**Pull request merged**